pkg/controllers/namespace/subnet_share_controller.go

```diff
@@ -0,0 +1,253 @@
+	err = r.createSubnetCRInK8s(ctx, subnetCR, subnetID)
+	if err != nil {
+		return err
+	}
+
+	namespacedName := types.NamespacedName{
+		Namespace: subnetCR.Namespace,
+		Name:      subnetCR.Name,
+	}
+	r.SubnetService.AddSharedSubnetToResourceMap(associatedName, namespacedName)
+
+	log.Info("Created Subnet CR for shared Subnet", "Namespace", ns, "Name", subnetCR.Name, "SharedSubnet", sharedSubnetPath)
+	return nil
+}
+
+// getExistingSharedSubnetCRs gets a map of existing shared Subnet CRs in the namespace
+func (r *NamespaceReconciler) getExistingSharedSubnetCRs(ctx context.Context, ns string) (map[string]*v1alpha1.Subnet, error) {
+	// Get shared subnets for the namespace from the SharedSubnetResourceMap
+	sharedSubnets := r.SubnetService.GetSharedSubnetsForNamespace(ns)
+
+	// Create a map of existing shared Subnet CRs
+	existingSharedSubnets := make(map[string]*v1alpha1.Subnet)
+
+	// For each associatedResource, get the Subnet CR
+	for associatedResource, namespacedNames := range sharedSubnets {
+		for _, namespacedName := range namespacedNames {
+			subnet := &v1alpha1.Subnet{}
+			err := r.Client.Get(ctx, namespacedName, subnet)
+			if err != nil {
+				log.Error(err, "Failed to get Subnet CR", "NamespacedName", namespacedName)
+				continue
+			}
+			existingSharedSubnets[associatedResource] = subnet
+			// We only need one Subnet CR per associatedResource
+			break
+		}
+	}
+
+	return existingSharedSubnets, nil
+}
+
+// processNewSharedSubnets creates Subnet CRs for new shared subnets
+func (r *NamespaceReconciler) processNewSharedSubnets(ctx context.Context, ns string,
+	vpcNetConfig *v1alpha1.VPCNetworkConfiguration, existingSharedSubnets map[string]*v1alpha1.Subnet) (map[string]*v1alpha1.Subnet, error) {
+
+	unusedSubnets := make(map[string]*v1alpha1.Subnet)
+	processedSubnets := make(map[string]bool)
+
+	for _, sharedSubnetPath := range vpcNetConfig.Spec.Subnets {
+		associatedResource, err := common.ConvertSubnetPathToAssociatedResource(sharedSubnetPath)
+		if err != nil {
+			log.Error(err, "Failed to convert Subnet path to associated resource", "Namespace", ns, "SharedSubnet", sharedSubnetPath)
+			return unusedSubnets, err
+		}
+
+		if _, exists := existingSharedSubnets[associatedResource]; !exists {
+			err := r.createSharedSubnetCR(ctx, ns, sharedSubnetPath)
+			if err != nil {
+				log.Error(err, "Failed to create Subnet CR for shared Subnet", "Namespace", ns, "SharedSubnet", sharedSubnetPath)
+				return unusedSubnets, err
+			}
+		}
+		processedSubnets[associatedResource] = true
+	}
+
+	for k, v := range existingSharedSubnets {
+		if !processedSubnets[k] {
+			unusedSubnets[k] = v
+		}
+	}
+
+	return unusedSubnets, nil
+}
+
+// checkSubnetReferences checks if a Subnet CR is referenced by any SubnetPort CRs or SubnetConnectionBindingMap CRs
+func (r *NamespaceReconciler) checkSubnetReferences(ctx context.Context, ns string, subnet *v1alpha1.Subnet, associatedResource string) (bool, error) {
```

Comment:
this func is to check if a Subnet CR has reference, associatedResource string param is not needed.